CoreFoundation/Base.subproj/CFAsmMacros.h

```diff
@@ -10,10 +10,22 @@
 #if !defined(__COREFOUNDATION_CFASMMACROS__)
 #define __COREFOUNDATION_CFASMMACROS__ 1
 
+#if DEPLOYMENT_RUNTIME_SWIFT
+#include <CoreFoundation/TargetConditionals.h>
+#else
+#include <TargetConditionals.h>
+#endif
+
 #define CONCAT(a,b) a##b
 #define CONCAT_EXPANDED(a,b) CONCAT(a,b)
 #define _C_LABEL(name) CONCAT_EXPANDED(__USER_LABEL_PREFIX__,name)
 
+// Define section name for read-only data section, if available.
+// TODO: equivalent section names for other operating systems.
+#if (TARGET_OS_LINUX || TARGET_OS_ANDROID || TARGET_OS_BSD) && defined(__ELF__)
```

Comment:
This feels wrong.  This section is really a property of the object file, not the OS.  I really prefer that we just conditionalise this on `__ELF__`, and sink this into the data files.

Reply:
Done, I agree that just `__ELF__` is probably enough. Thanks for taking a look.